Automattic/jetpack
# Payment Blocks: Display the product manager on free sites connected to Stripe

#### Changes proposed in this Pull Request:

* Only hide the payment blocks' product manager on free sites without an active Stripe connection.

There is an edge case where free sites can have an active Stripe connection, if they were once paid sites using Stripe, and then downgraded to free.
WordPress.com explicitly supports this edge case, by raising the commissions on transactions, to avoid breaking existing payment blocks and active subscriptions.

The product manager component, though, was configured to only be displayed on paid sites.
The unintended consequence was that free sites with a Stripe connection didn't have any way to check the payment plan associated to a block, change it if needed, or receive warning messages if any.

This PR makes sure that the product manager component is displayed if the site is paid, or if there is an active Stripe connection.

The only difference is that free sites are not allowed to create new products, so the creation form is not shown to them.

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:






* On a paid site with Stripe not connected:
  * Insert a payment block and make sure there are no regressions.
  * You get the "connect to stripe" button.
  * After connecting, you get the products selector, and can create new products.
* On a free site with Stripe connected:
  * Insert a payment block and make sure you get the product selector.
  * Make sure you cannot create new products: no "Add new" button in the toolbar selector, no "Add new" form in the block sidebar.
* On a free site with Stripe disconnected:
  * Insert a payment block and make sure you don't get the product selector at all.
